### PR TITLE
New Solutions sweep counting annotator

### PR DESCRIPTION
**Pull request by RizwanMunawar**
@glenn-jocher Hi! I’ve added a new annotation feature that allows users to visualize objects dynamically based on a sweep line, which can be controlled directly with the mouse. The objects are annotated according to the sweep line's position. This feature currently supports segmentation and detection models, but we can extend it to work with pose estimation models as well. Thanks🙂

### Demo 🚀
https://github.com/user-attachments/assets/f4ebac26-3672-4506-8ac4-0b5bfa54d324

### Docs Display
<p align="center">
  Screenshot 1
  Screenshot 2
</p>

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary  
This PR introduces a "Sweep Annotation" utility in Ultralytics for video annotation with enhanced tracking and visualization features. 🚀  

### 📊 Key Changes  
- Added an **example for a Sweep Annotation** workflow in the documentation, showcasing how to annotate and track objects in videos. 🖍️  
- Introduced a `sweep_annotator` function in `plotting.py` to draw a dynamic sweep line and display the count of objects crossing a line. 🚦  
- Enhanced existing utilities (`seg_bbox`) to support better annotation customization.  

### 🎯 Purpose & Impact  
- **Improved Annotation Capabilities**: The "Sweep Annotation" allows users to track objects crossing a threshold in real-time, which is useful for applications like traffic monitoring or event detection. 🎥  
- **Simplified Implementation**: The detailed example makes it easier for developers to use these tools straight out of the box, reducing the learning curve. 📚  
- **Enhanced Versatility**: The new function adds flexibility to the annotation pipeline, opening possibilities for novel use cases. 🌟

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @RizwanMunawar, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response to help streamline our review process. An Ultralytics engineer will also take a closer look at your submission soon. In the meantime, please ensure the following checklist is complete for a smooth integration:

- ✅ **Define a Purpose**: Clearly state the purpose of your feature or fix in your PR description. Link back to any [relevant issues](https://github.com/ultralytics/ultralytics/issues) to provide additional context. Your commit messages should be clear, concise, and follow the project's conventions.
- ✅ **Synchronize with Source**: Make sure your PR is up-to-date with the `ultralytics/ultralytics` `main` branch. You can update it by clicking the 'Update branch' button on GitHub or by using `git pull` and `git merge main` from the command line.
- ✅ **Ensure CI Checks Pass**: All Continuous Integration (CI) checks must be passing. You can find more about our [CI](https://docs.ultralytics.com/help/CI/) process in the documentation. If any checks are failing, please address these before the review.
- ✅ **Update Documentation**: Update our [documentation](https://docs.ultralytics.com) where necessary to reflect any new or changed features.
- ✅ **Add Tests**: Include or update tests that cover your changes to ensure they are adequately tested. Verify that all tests pass.
- ✅ **Sign the CLA**: If this is your first PR to Ultralytics, please make sure to sign our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) by stating "I have read the CLA Document and I sign the CLA" in a new comment.
- ✅ **Minimize Changes**: Ensure your changes are the **minimal** required to achieve your intended fix or feature addition. As Bruce Lee wisely said, _"It is not daily increase but daily decrease, hack away the unessential."_

For detailed guidance, feel free to consult our [Contributing Guide](https://docs.ultralytics.com/help/contributing). If you encounter any issues or have questions, do not hesitate to comment below.

Thank you for your contribution to Ultralytics! 🎉🚀

**Comment by glenn-jocher** [comment]
@RizwanMunawar oh wow, this is really cool!

File: docs/en/usage/simple-utilities.md [hunk 1]
```diff
@@ -374,6 +374,157 @@ See docstring for each function or visit the `ultralytics.utils.ops` [reference
 
 Ultralytics includes an Annotator class that can be used to annotate any kind of data. It's easiest to use with [object detection bounding boxes](../modes/predict.md#boxes), [pose key points](../modes/predict.md#keypoints), and [oriented bounding boxes](../modes/predict.md#obb).
 
+#### Ultralytics Sweep Annotation
+
+!!! example "Python Examples using YOLO11 🚀"
+
+    Just set your video path and model file to get started :) great for trying out custom models for inference.
+
+    🚀 Also, explore sweeping annotation visuals and per-frame counting in action!"
```
Comment (glenn-jocher) [inline docs/en/usage/simple-utilities.md:383]: Let's keep this professional, no emojis or smily faces please.

**Comment by glenn-jocher** [comment]
@RizwanMunawar there's significant duplication of code here between the box and segment example. Let's use one only please.

**Comment by RizwanMunawar** [comment]
@glenn-jocher All set! I've updated the code to ensure a single Python script seamlessly supports both Segmentation and Detection models. Thanks!

**Comment by RizwanMunawar** [comment]
@glenn-jocher Hi, I’ve resolved all the conflicts identified by the Ultralytics assistant, and everything looks good now. Could you please review it? If it’s all set, we can proceed with the merge. Thanks!

**Comment by glenn-jocher** [comment]
@RizwanMunawar looks good, PR merged!

**Pull request merged**